File: 프로그래머스/1/42840. 모의고사/모의고사.py
#12345
#2, 1, 2, 3, 2, 4, 2, 5, 
#3, 3, 1, 1, 2, 2, 4, 4, 5, 5,
def solution(answers):
    answer = []
    num1 = [1, 2, 3, 4, 5]
    num2 = [2, 1, 2, 3, 2, 4, 2, 5]
    num3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    cnt_1 = 0
    cnt_2 = 0
    cnt_3 = 0
    for i in range(len(answers)):
        if num1[i%len(num1)] == answers[i]: cnt_1+=1
        if num2[i%len(num2)] == answers[i]: cnt_2+=1
        if num3[i%len(num3)] == answers[i]: cnt_3+=1
    if cnt_1 < cnt_2:
        if cnt_2 < cnt_3:
            answer.append(3)
        elif cnt_2 > cnt_3:
            answer.append(2)
        else:
            answer.append(2)
            answer.append(3)
    elif cnt_1 > cnt_2:
        if cnt_1 < cnt_3:
            answer.append(3)
        elif cnt_1 > cnt_3:
            answer.append(1)
        else:
            answer.append(1)
            answer.append(3)
    else:
        if cnt_1 < cnt_3:
            answer.append(3)
        elif cnt_1 > cnt_3:
            answer.append(1)
            answer.append(2)
        else:
            answer.append(1)
            answer.append(2)
            answer.append(3)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.ArrayList;
class Solution {
    public int[] solution(int[] answers) {
        int[] answer = new int[3];
        //1, 2, 3, 4, 5
        //2, 1, 2, 3, 2, 4, 2, 5
        //3, 3, 1, 1, 2, 2, 4, 4, 5, 5
        int[] p1 = {1, 2, 3, 4, 5};
        int[] p2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] p3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] cnt = {0,0,0};
        for(int i = 0; i < answers.length; i++) {
            
            if(p1[i%p1.length] == answers[i]) {
                cnt[0]++;
            } if(p2[i%p2.length] == answers[i]) {
                cnt[1]++;
            } if(p3[i%p3.length] == answers[i]) {
                cnt[2]++;
            }
        }
        int max = Math.max(cnt[0], Math.max(cnt[1], cnt[2]));
        
        ArrayList<Integer> lst = new ArrayList<>();
        for (int i=0; i<3;i++) {
            if(max == cnt[i]) {
                lst.add(i+1);
            }
        }
   
        
        return lst.stream().mapToInt(i->i).toArray();
    }
}